Icon aliases as class constants

nolannordlund opened the pull request:
## Description

This PR adds class constants for all icon aliases, similar to `PanelsRenderHook` and `TablesRenderHook`.

## Visual changes

N/A

## Functional changes

- [x] Code style has been fixed by running the `composer cs` command.
- [x] Changes have been tested to not break existing functionality.
- [x] Documentation is up-to-date.

nolannordlund committed:
Added icon alias constants

nolannordlund committed:
Updating references

nolannordlund committed:
Updated instances of resolve

nolannordlund committed:
Rename and document chart widget filter button

nolannordlund commented:
I found a previously undocumented icon alias, `widgets::chart-widget.filter`. It didn't seem to match previous patterns since there were no other `widgets::` icon alias. There were some `widgets` aliases with the `panels::` namespace however so I decided to rename it.

nolannordlund committed:
Updating more aliases

nolannordlund committed:
Updating more icon aliases

nolannordlund committed:
Updating a few more aliases

nolannordlund committed:
Adding rich editor icon aliases

nolannordlund commented:
Found a few more undocumented and slightly off-pattern aliases:
- `forms:components.rich-editor.panels.custom-blocks.close-button`
- `forms:components.rich-editor.panels.custom-block.delete-button`
- `forms:components.rich-editor.panels.custom-block.edit-button`
- `forms:components.rich-editor.panels.merge-tags.close-button`

I changed them to all have double semicolons and added them to `FormsIconAlias`. They have also been documented

nolannordlund committed:
Style fix

nolannordlund committed:
Correcting namespaces and file locations

nolannordlund committed:
Styling

danharrin commented:
> I found a previously undocumented icon alias, `widgets::chart-widget.filter`. It didn't seem to match previous patterns since there were no other `widgets::` icon alias. There were some `widgets` aliases with the `panels::` namespace however so I decided to rename it to `panels::widgets.chart-widget.filter` and added the constant in `PanelsIconAlias`

Please revert this, the other widgets are included in the panels package only but this is the only alias that exists for widgets, even outside of panels.

nolannordlund committed:
Moved alias to widgets namespace

nolannordlund commented:
I reverted the string to `widgets::chart-widget.filter` but I added it to a new `WidgetsIconAlias` class as `CHART_WIDGET_FILTER`. I added a section to the docs for it also. If you want all of that reverted, just let me know and I'll change it is simply be as it was before.
